Userland/Libraries/LibPDF/Parser.cpp

```diff
@@ -1023,7 +1024,7 @@ RefPtr<StreamObject> Parser::parse_stream(NonnullRefPtr<DictObject> dict)
     ReadonlyBytes bytes;
 
     auto maybe_length = dict->get(CommonNames::Length);
-    if (maybe_length.has_value()) {
+    if (maybe_length.has_value() && (!maybe_length->is_ref() || m_xref_table)) {
```

Comment:
I assume the logic here is that we can get here from `initialize_linearization_dict` due to the `parse_indirect_value` call, and as far as I can see this is the only way `m_xref_table` can be null here. But if it is in fact null when we get here, we should just `return {}` as we don't want to parse a stream anyways.

Reply:
Indeed, `parse_stream` is called before the `m_xref_table` is populated because you are trying to read the first object after the file header and in this case (`non-linearized.pdf`) the first object is a `stream`, so later down in the path a assert fail because the `m_xref_table` is not there.

Reply:
I was suggesting a change here. We should return early if the xref table isn't defined because we don't want to actually parse a stream while reading the linearization dictionary.

Reply:
if we return `{}` from `parse_stream`, then we can't have this code return `Error`
```cpp
    auto dict_value = m_document->resolve(parse_indirect_value());
    if (!dict_value || !dict_value.is_object())
        return LinearizationResult::Error;
```
because `parse_indirect_value` would return `{}` making `dict_value` fail. 

Do you see any other option?

Reply:
Ahh, right, makes sense. Not too big of a deal then, it'd be nice to have but is certainly out of the scope of this PR